crates/aptos/src/config/mod.rs

```diff
@@ -236,7 +236,7 @@ fn find_workspace_config(
                 current_path.push(CONFIG_FOLDER);
                 if current_path.is_dir() {
                     break Ok(current_path);
-                } else if !current_path.pop() && !current_path.pop() {
+                } else if !(current_path.pop() && current_path.pop()) {
```

Comment:
This changes the original condition here. Was this the root cause?

Reply:
if I give you "/a/b/c/" and you call pop that will return true which when inverted returns false, so you never get to the second pop and you end up in an infinite loop.

Reply:
Yeah this was my mistake good catch